controllers/userStatus.js

```diff
@@ -240,6 +240,31 @@ const updateUserStatusController = async (req, res, next) => {
   }
 };
 
+const syncUserStatus = async (req, res, next) => {
+  try {
+    await updateAllUserStatus(req, res, next);
+    const usersData = await getTaskBasedUsersStatus(req, res, next);
```

Comment:
IMO logic that should be part of controllers should not be used as middleware
> also, these functions seem to be returning res, which will result in an error, because in that case we are calling res/json 3 times.

Appologies did not understand